src/values.rs

```diff
@@ -309,14 +310,106 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::collections::hash_map::RandomState;
+
+    macro_rules! assert_empty {
+        ($x:expr) => {
+            assert!($x.is_empty());
+            assert_eq!($x.len(), 0);
+            assert_eq!($x.iter().count(), 0);
+            assert_eq!($x.into_iter().count(), 0);
+            assert_eq!($x.get_one(), None);
+        };
+    }
+
+    macro_rules! assert_len {
+        ($x:expr, $n:expr) => {
+            assert!(!$x.is_empty());
+            assert_eq!($x.len(), $n);
+            assert_eq!($x.iter().count(), $n);
+            assert_eq!($x.into_iter().count(), $n);
+        };
+    }
+
+    fn has_short_values<T, S>(v: &Values<T, S>) -> bool {
```

Comment:
I don't have to do this this way. I could add them in the `impl` outside the test module. I could also add them to an `impl` inside the test module so they're only available for tests.

I could also, instead of checking `v.0` in this method, I could use the public `v.iter()` in a test to do like:
```rust
match v.iter() {
  ValuesIter::Short(_) => {},
  _ => panic!("Does not have short values!"),
}
```
because `panic!` and `assert!` are basically the same. Lemme know which you'd prefer 👍 (or whatever other alternative!)

Reply:
I think a method on `Values` annotated with `#[cfg(test)]` is probably the way to go here

Reply:
Also, you can simplify this implementation to:
```rust
fn is_short(&self) -> bool {
  matches!(self.0, ValuesInner::Short(_))
}
```

Reply:
Ooo TIL! I was looking for something like this!

Reply:
ac8fc8ca039225c21057adc41f83e6015879f600